### PR TITLE
Provide custom palette directly to neogit

### PR DESCRIPTION
**Pull request by gldtn**
Better integration with Neogit custom palette: https://github.com/NeogitOrg/neogit/blob/0820552997dd4dda4faa147228f0e61ea88290a3/doc/neogit.txt#L563

### CODE REVIEW

File: lua/cyberdream/extensions/neogit.lua [hunk 1]
```diff
@@ -5,28 +5,34 @@ local M = {}
 --- @param t CyberdreamPalette
 function M.get(opts, t)
     opts = opts or {}
+    local neogit = require("neogit")
```
Comment (scottmckendry) [inline lua/cyberdream/extensions/neogit.lua:8]: We need to check if `neogit` can be loaded before we try to require it. If you don't have the plugin installed, it breaks cyberdream:

```
Failed to run `config` for cyberdream.nvim

vim/_editor.lua:0: /home/scott/git/nix/nvim/init.lua..nvim_exec2() called at /home/scott/git/nix/nvim/init.lua:0../home/scott/git/cyberdream.nvim/colors/cyberdream.lua: Vim(colorscheme):E5113: Error while calling lua chunk: ...git/cyberdream.nvim/lua/cyberdream/extensions/neogit.lua:8: module 'neogit' not found:
	no field package.preload['neogit']
```

```suggestion
    if not package.loaded["neogit"] then
        return {}
    end
    local neogit = require("neogit")
```

**Review by scottmckendry — CHANGES_REQUESTED** [review]

**Comment by scottmckendry** [comment]
You could use Cattpuccin's implementation as a reference:

https://github.com/catppuccin/nvim/blob/f8a155ab5891c5d2fb709b7e85627f1783d5a5d9/lua/catppuccin/groups/integrations/neogit.lua

You probably won't need to define _all_ highlight groups; most default links should be fine to leave as is. 

They use `darken` & `lighten` helpers for darker and lighter colours for backgrounds etc. 

If you want to do something similar you can use `blend`. There's an example of this here:

https://github.com/scottmckendry/cyberdream.nvim/blob/731290012f435e06e4d5d64ed7c09cec49df2663/lua/cyberdream/extensions/markdown.lua

**Comment by gldtn** [comment]
This removes `.setup()` call. Overall Neogit should look much cleaner with this update.

Question, when you set Cyberdream with `transparent = true`, does `fg` becomes white if set to `t.bg` on your terminal? Or is this a normal behavior in Neovim? I'm assuming Cyberdream sets `bg` to `"NONE"` if transparency is enabled.

For example:
`NeogitCommitViewHeader = { bg = t.cyan, fg = t.bg }`

To avoid this I set it to `t.bg_solid`

I noticed this behavior when customizing my fzf-lua/telescope, I can open a issue to show screenshots if necessary.

Thanks!

**Comment by scottmckendry** [comment]
Great stuff @gldtn 🙂 thanks again for contributing to the project!

**Pull request merged**